crates/server/src/packet_handlers/interaction.rs

```diff
@@ -1,35 +1,75 @@
 use crate::{ClientId, NetworkId, Server};
-use common::events::InteractEntityEvent;
 use common::Game;
-use ecs::{Entity, EntityRef, SysResult};
-use libcraft_core::{InteractHand, InteractionType, Vec3f};
+use common::{
+    events::{BlockInteractEvent, BlockPlacementEvent, InteractEntityEvent},
+    interactable::InteractableRegistry,
+};
+use ecs::{Entity, SysResult};
+use libcraft_core::{BlockFace as LibcraftBlockFace, Hand, InteractionType, Vec3f};
 use protocol::packets::client::{
     BlockFace, InteractEntity, InteractEntityKind, PlayerBlockPlacement, PlayerDigging,
     PlayerDiggingStatus,
 };
 
 /// Handles the player block placement packet. Currently just removes the block client side for the player.
 pub fn handle_player_block_placement(
-    game: &Game,
-    server: &mut Server,
+    game: &mut Game,
+    _server: &mut Server,
     packet: PlayerBlockPlacement,
-    player: EntityRef,
+    player: Entity,
 ) -> SysResult {
-    let position = match packet.face {
-        BlockFace::Bottom => packet.position.down(),
-        BlockFace::Top => packet.position.up(),
-        BlockFace::North => packet.position.north(),
-        BlockFace::South => packet.position.south(),
-        BlockFace::West => packet.position.west(),
-        BlockFace::East => packet.position.east(),
+    let hand = match packet.hand {
+        0 => Hand::Main,
+        1 => Hand::Offhand,
+        _ => unreachable!(),
     };
 
-    log::trace!("Got player block placement at {:?}", position);
+    let face = match packet.face {
+        BlockFace::North => LibcraftBlockFace::North,
+        BlockFace::South => LibcraftBlockFace::South,
+        BlockFace::East => LibcraftBlockFace::East,
+        BlockFace::West => LibcraftBlockFace::West,
+        BlockFace::Top => LibcraftBlockFace::Top,
+        BlockFace::Bottom => LibcraftBlockFace::Bottom,
+    };
+
+    let cursor_position = Vec3f::new(
+        packet.cursor_position_x,
+        packet.cursor_position_y,
+        packet.cursor_position_z,
+    );
+
+    let block_kind = { game.block(packet.position).unwrap().kind() };
```

Comment:
This could panic if the block is unloaded, so we should have a check here.

Reply:
I'm just gonna kick the player if they do that then.